common/aws/test/fragment_test.go

```diff
@@ -0,0 +1,330 @@
+	assert.Equal(t, expectedFragmentCount, fragmentCount)
+
+	// Test a file that is one more than a multiple of the fragment size
+	fragmentSize = rand.Intn(100) + 100
+	expectedFragmentCount = rand.Intn(10) + 2
+	fileSize = fragmentSize*(expectedFragmentCount-1) + 1
+	fragmentCount = s3.GetFragmentCount(fileSize, fragmentSize)
+	assert.Equal(t, expectedFragmentCount, fragmentCount)
+}
+
+// Fragment keys take the form of "prefix/body-index[f]". Verify the prefix part of the key.
+func TestPrefix(t *testing.T) {
+	tu.InitializeRandom()
+
+	keyLength := rand.Intn(10) + 10
+	key := tu.RandomString(keyLength)
+
+	for i := 0; i < keyLength*2; i++ {
+		fragmentCount := rand.Intn(10) + 10
+		fragmentIndex := rand.Intn(fragmentCount)
+		fragmentKey, err := s3.GetFragmentKey(key, i, fragmentCount, fragmentIndex)
+		assert.NoError(t, err)
+
+		parts := strings.Split(fragmentKey, "/")
+		assert.Equal(t, 2, len(parts))
+		prefix := parts[0]
+
+		if i >= keyLength {
+			assert.Equal(t, key, prefix)
+		} else {
+			assert.Equal(t, key[:i], prefix)
+		}
+	}
+}
+
+// Fragment keys take the form of "prefix/body-index[f]". Verify the body part of the key.
+func TestKeyBody(t *testing.T) {
+	tu.InitializeRandom()
+
+	for i := 0; i < 10; i++ {
+		keyLength := rand.Intn(10) + 10
+		key := tu.RandomString(keyLength)
+		fragmentCount := rand.Intn(10) + 10
+		fragmentIndex := rand.Intn(fragmentCount)
+		fragmentKey, err := s3.GetFragmentKey(key, rand.Intn(10), fragmentCount, fragmentIndex)
+		assert.NoError(t, err)
+
+		parts := strings.Split(fragmentKey, "/")
+		assert.Equal(t, 2, len(parts))
+		parts = strings.Split(parts[1], "-")
+		assert.Equal(t, 2, len(parts))
+		body := parts[0]
+
+		assert.Equal(t, key, body)
+	}
+}
+
+// Fragment keys take the form of "prefix/body-index[f]". Verify the index part of the key.
+func TestKeyIndex(t *testing.T) {
+	tu.InitializeRandom()
+
+	for i := 0; i < 10; i++ {
+		fragmentCount := rand.Intn(10) + 10
+		index := rand.Intn(fragmentCount)
+		fragmentKey, err := s3.GetFragmentKey(tu.RandomString(10), rand.Intn(10), fragmentCount, index)
+		assert.NoError(t, err)
+
+		parts := strings.Split(fragmentKey, "/")
+		assert.Equal(t, 2, len(parts))
+		parts = strings.Split(parts[1], "-")
+		assert.Equal(t, 2, len(parts))
+		indexStr := parts[1]
+		assert.True(t, strings.HasPrefix(indexStr, fmt.Sprintf("%d", index)))
+	}
+}
+
+// Fragment keys take the form of "prefix/body-index[f]".
+// Verify the postfix part of the key, which should be "f" for the last fragment.
+func TestKeyPostfix(t *testing.T) {
+	tu.InitializeRandom()
+
+	segmentCount := rand.Intn(10) + 10
+
+	for i := 0; i < segmentCount; i++ {
+		fragmentKey, err := s3.GetFragmentKey(tu.RandomString(10), rand.Intn(10), segmentCount, i)
+		assert.NoError(t, err)
+
+		if i == segmentCount-1 {
+			assert.True(t, strings.HasSuffix(fragmentKey, "f"))
+		} else {
+			assert.False(t, strings.HasSuffix(fragmentKey, "f"))
+		}
+	}
+}
+
+func TestExampleInGodoc(t *testing.T) {
```

Comment:
Is the goal for this function to appear in godoc? I think for that to happen the function name has to start with `Example` and not `Test`: https://go.dev/blog/examples

Reply:
I give an example in the godoc of the function `getFragmentKey()`. Updated the godoc to mention that:

```
// TestExampleInGodoc tests the example provided in the documentation for getFragmentKey().
//
// Example: fileKey="abc123", prefixLength=2, fragmentCount=3
// The keys will be "ab/abc123-0", "ab/abc123-1", "ab/abc123-2f"
```